src/Common/BufferAllocationPolicy.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "config.h"
+
+#include "logger_useful.h"
+
+#include <list>
+
+namespace DB
+{
+
+struct BufferAllocationSettings
```

Comment:
If we move this struct to the `IBufferAllocationPolicy` class we can rename it to just `Settings`.

Reply:
addressed it